README.md

```diff
@@ -1,5 +1,38 @@
-# lighthouse 
-Controller to facilitate DNS discovery in multi-cluster connected environments (proof-of-concept state)
+# Lighthouse 
+The lighthouse along with Submariner facilitate DNS discovery in multi-cluster connected environments. The solution aims to compatible with any CNI-compatible cluster network provider.
+
+# Architecture
+The below digram shows the basic lighthouse architecture.
+
+![Lighthouse architecture](https://raw.githubusercontent.com/submariner-io/lighthouse/master/docs/img/architecture.png) 
+
+## Lighthouse Controller
+There is a central service discovery controller that gathers the information from the clusters, decides what information is to be shared and distributes the information as newly defined CRDs.
+
+The lighthouse controller will be running along with the admiral control plane. It uses the API’s exposed by the admiral to connect to each cluster and listen for service CRUD. Once it is notified that a new service is created and if the service needs to be distributed, it will frame the lighthouse CRD and the same will be distributed to all the clusters that are part of the control plane. The flow for update and delete will be similar.
```

Comment:
Use future or present throughout document, don't mix two. Better stick to present. "Lighthouse controller runs along with admiral...".

Link to Admiral at first mention.

Reply:
Done.